src/Controller/Api/MediaController.php

```diff
@@ -406,24 +392,11 @@ protected function handleWriteMedium(Request $request, MediaInterface $media, Me
 
             $view = FOSRestView::create($media);
 
-            // BC for FOSRestBundle < 2.0
-            if (method_exists($view, 'setSerializationContext')) {
-                $serializationContext = SerializationContext::create();
-                $serializationContext->setGroups(['sonata_api_read']);
-                $serializationContext->enableMaxDepthChecks();
-                $view->setSerializationContext($serializationContext);
-            } else {
-                $context = new Context();
-                $context->setGroups(['sonata_api_read']);
-
-                // NEXT_MAJOR: simplify when dropping FOSRest < 2.1
-                if (method_exists($context, 'disableMaxDepth')) {
-                    $context->disableMaxDepth();
-                } else {
-                    $context->setMaxDepth(0);
-                }
-                $view->setContext($context);
-            }
+            $context = new Context();
```

Comment:
Is this even needed, can't it be replaced with `View` Annotation?

Reply:
I am not sure about that, It was originally implemented like that. Do you know a better way? I am not a user of FOSRest atm, just doing the cleanup

Reply:
If I am correct, annotation [here](https://github.com/jordisala1991/SonataMediaBundle/blob/79952066c3f9dac4d227f05ac55dca943b196b44/src/Controller/Api/GalleryController.php#L85 )

`@View(serializerGroups={"sonata_api_read"}, serializerEnableMaxDepthChecks=true)`

Can replace:

```
            $view = FOSRestView::create($gallery);
            $context = new Context();
            $context->setGroups(['sonata_api_read']);
            $context->disableMaxDepth();
            $view->setContext($context);
            return $view;
```

to just 

```
return $gallery;
```

Only change `serializerEnableMaxDepthChecks` to `false`.

Reply:
Hopefully, someone else is using FOSRest and could confirm this.

Reply:
But on our case we are doing this only on the `if`, on the else part we are returning a form

Reply:
That does not affect form, form is not valid it returns validation errors, maybe I am using FOSRest wrong but I have it like that everywhere :smile:

Reply:
is this what you had in mind?

Reply:
yes, but someone needs to test/confirm this will work.